modules/standard/Regex.chpl

```diff
@@ -585,6 +585,13 @@ proc bytes.this(m:regexMatch) {
   else return b"";
 }
 
+
+pragma "no doc"
+record __serializeHelper {
+  var home:locale;
+  var _regex:qio_regex_t;
+}
```

Comment:
the actual copy gets done in `chpl__deserialize` and we need a way to do `on serialized.home`

Reply:
That's surprising to me. I think in the ideal case we're just bundling the options/string into the record on the initiating side, sending them along with the AM/on-stmt, and then unbundling on the remote side with no added communication. I think you're saying that now we have another on-stmt in deserialize to go and grab the options, but it's not clear to me why that's needed instead of just packing the options into serializeHelper.

Reply:
Oh, and I guess I misread both the code and your comment. You're saying we have an on-stmt to grab the options and string, and the __serializeHelper contains the compiled regex? But I thought we can't bit-copy that?

Reply:
> I think in the ideal case we're just bundling the options/string into the record

Yeah totally agree. That was the original implementation and this approach was the first fix I thought of (the problem being that referencing `this` inside the `on this.home {...}` in `chpl__serialize` was triggering `this` to get `rvf` which called `chpl__serialize` again in a loop). But now I think I see a better way to do it.